### PR TITLE
Workaround for GitHub search API bug 

### PR DESCRIPTION
**Pull request by PhilipTrauner**
If an issue title contains a number that is surrounded by two spaces, the `in:title` search qualifier is ignored, and the `number` property of issues is also taken into account. Issues that are returned based on said `number` property also always have a `score` of 100, which places them before the actual issue.

To work around this limitation, the `og:title` issue lookup function now iterates over all returned issues and returns the first one that exactly matches the OpenGraph title.

### CODE REVIEW

**Comment by RohanNagar** [comment]
Linking #119 here. @jdanyow could you give this a review?

**Comment by PhilipTrauner** [comment]
Any update on this @jdanyow? 🙂

File: src/github.ts [hunk 1]
```diff
@@ -139,14 +139,12 @@ export function loadIssueByTerm(term: string) {
...
     return response.json();
   }).then(results => {
-    if (results.total_count === 0) {
-      return null;
-    }
-    if (results.total_count > 1) {
-      // tslint:disable-next-line:no-console
-      console.warn(`Multiple issues match "${q}". Using earliest created.`);
+    for (const item of results.items) {
+      if (item.title === term) {
```
Comment (jdanyow) [inline src/github.ts]: Would changing this line to use indexOf still fix your issue? I'm concerned about the impact this change will have because the [config documentation](https://utteranc.es/#heading-mapping) say "issue title **contains** term" rather than "**equals** term".

```diff
- if (item.title === term) {
+ if (item.title.toLowerCase().indexOf(term.toLowerCase()) !== -1) {
```

Comment (PhilipTrauner) [reply src/github.ts]: I supposed that this could also work, but I'm not sure if there aren't potential edge cases. I'll apply your suggestion regardless, because it will fix the initial issue.

**Comment by PhilipTrauner** [comment]
Any update on this @jdanyow? 🙂

**Comment by jdanyow** [comment]
aiming to ship an update with this fix in the next day or two

**Comment by jdanyow** [comment]
@PhilipTrauner / @RohanNagar I've shipped an update using the code from this PR with a slight modification: instead of returning null at the very end I kept the logic that would return the first match.

I want to do some more testing before returning null when none of the items in the issue search response match on title.

**Comment by PhilipTrauner** [comment]
I fully understand why you'd want to err on the side of caution with this one, but the fix you deployed doesn't resolve the actual problem this PR was meant to address. Posts that contain issue identifiers for which no comment issue has been created yet *still* resolve to the wrong issue (e.g. https://philip-trauner.me/blog/post/pedantic-elgato-thunderbolt-3-pro-dock-review).

**Comment by jdanyow** [comment]
@PhilipTrauner I agree, this isn't fully fixed yet. I've kept #119 open.
